### DISCUSSION TITLE
what is the difference between tangyoha version and Dineshkarthik version ?  i am confused

### DISCUSSION THREAD

**Discussion by azerfr2** [tangyoha/telegram_media_downloader#122] [Q&A] [answered]
how to know the differences ?

**Comment by tangyoha** [comment, marked as answer]
> how to know the differences ?

check release page

**Comment by tangyoha** [comment]
if you like this project, please give a star!

**Comment by tangyoha** [comment]
Hello! Great question about the differences between tangyoha and Dineshkarthik versions.

**Key Differences**:

**tangyoha/telegram_media_downloader** (this repo):
- Fork of the original Dineshkarthik version
- Added Web GUI for monitoring download progress (port 5050)
- Bot mode support for receiving forwarded messages
- Docker support with better configuration
- Rclone integration for automatic cloud upload
- More active maintenance and Chinese language support
- Additional features like download filters, file size limits
- Better support for private groups and restricted content

**Dineshkarthik/telegram_media_downloader** (original):
- Original project, more basic functionality
- Command-line only interface
- Simpler configuration
- Less frequent updates

**Which one to choose**:
- Use **tangyoha version** if you want:
  - Web interface for monitoring
  - Bot mode for easy forwarding
  - Docker deployment
  - Cloud upload integration
  - More features and active development

- Use **Dineshkarthik version** if you want:
  - Simpler, more lightweight tool
  - Original stable version
  - Minimal dependencies

The tangyoha version is essentially an enhanced fork with many additional features while maintaining compatibility with the core functionality.
